modules/substrate/src/lib.rs

```diff
@@ -361,6 +362,38 @@ impl<T: Config> Module<T> {
 	}
 }
 
+impl<T: Config> bp_header_chain::HeaderChain<BridgedHeader<T>> for Module<T> {
+	fn best_finalized() -> BridgedHeader<T> {
+		PalletStorage::<T>::new().best_finalized_header().header
+	}
+
+	fn authority_set() -> AuthoritySet {
+		PalletStorage::<T>::new().current_authority_set()
+	}
+
+	fn import_header(header: BridgedHeader<T>) -> Result<(), ()> {
+		let mut verifier = verifier::Verifier {
+			storage: PalletStorage::<T>::new(),
+		};
+
+		let _ = verifier.import_header(header.hash(), header).map_err(|_| ())?;
+
+		Ok(())
+	}
+
+	fn import_finality_proof(header: BridgedHeader<T>, finality_proof: Vec<u8>) -> Result<(), ()> {
+		let mut verifier = verifier::Verifier {
+			storage: PalletStorage::<T>::new(),
+		};
+
+		let _ = verifier
+			.import_finality_proof(header.hash(), finality_proof.into())
```

Comment:
Just to be sure that it is known - this fn verifies justification, but you have already did that in `submit_finality_proof`. Is it left for future PRs?

Reply:
Yeah I know it's being verified twice 😅

I would like to change this in the future so that we could import the headers and finality proofs directly without additional verification